### ISSUE TITLE
[Workflow] Map registered workflow types

### ISSUE THREAD

**Issue by hhunter-ms** [dapr/docs#3317]
We'll need to make the following documentation changes:

- [ ] The workflow architecture doc needs to be updated to reflect the 1:1 mapping between workflow types and internal actor types.
- [ ] The workflow development guide needs to mention the workflow and activity name restrictions.
- [ ] A sample should be added that demonstrates cross-app workflow scheduling.

Related issues:
- Proposal: dapr/dapr#6073
- Issue: dapr/dapr#6013

@cgillum and @nyemade-uversky for visibility

**Comment by cgillum** [comment]
> The workflow architecture doc needs to be updated to reflect the 1:1 mapping between workflow types and internal actor types.

We made a decision to not do 1:1 mapping between workflow and internal actor types. However, we did change the naming convention for these actor types, so I will update that information in the architecture doc.

> The workflow development guide needs to mention the workflow and activity name restrictions.

We don't have any name restrictions, but we do have new instance ID restrictions in v1.11. I'll update the documentation to reflect these restrictions.

> A sample should be added that demonstrates cross-app workflow scheduling.

This scenario is no longer possible as originally envisioned. Cross-app interactions are still technically possible using the Dapr client as an intermediary, but I think this kind of sample is best done a separate PR since it's a bit advanced.
